src/Databases/PostgreSQL/DatabaseMaterializedPostgreSQL.cpp

```diff
@@ -521,7 +521,8 @@ void registerDatabaseMaterializedPostgreSQL(DatabaseFactory & factory)
 
         if (auto named_collection = tryGetNamedCollectionWithOverrides(engine_args, args.context))
         {
-            configuration = StoragePostgreSQL::processNamedCollectionResult(*named_collection, args.context, false);
+            PostgreSQLSettings postgresql_settings;
```

Comment:
We can support settings for materialized postgresql as well. It already has its own settings, but we can have a union of settings, see example https://github.com/ClickHouse/ClickHouse/blob/06e7def57ba2658ed32294ac10b98b00fa776d6e/src/Storages/RabbitMQ/RabbitMQSettings.cpp#L51-L54.
E.g. here we have a union of rabbitmq settings + rabbitmq obsolete settings + format settings. So it will require to just add a small modification to the settings define statement.